### PR TITLE
Add Husky and lint-staged

### PR DESCRIPTION
**Pull request by AdriSolid**
Story details: https://app.clubhouse.io/cartoteam/story/127479

It needed to integrate the default CRA ESLint config. ESLint aligned on lib and template.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+indent_style = space
+insert_final_newline = true
+max_line_length = 80
+trim_trailing_whitespace = true
+
+[*.md]
+max_line_length = 0
+trim_trailing_whitespace = false
+
+[*.ejs*]
```
Comment (neokore) [inline .editorconfig:18]: I think that this is not needed

Comment (AdriSolid) [reply .editorconfig:18]: We should remove it from template repo also, as it has same editorconfig file

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "printWidth": 90,
+  "arrowParens": "always",
+  "semi": true,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "jsxSingleQuote": true
```
Comment (neokore) [inline .prettierrc:7]: Lots of :heart: here!
Just a question, can we add `trailingCommas: none`? What are your feelings about trailing commas?

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "printWidth": 90,
```
Comment (neokore) [inline .prettierrc:2]: For me maybe 90 chars is a bit tight, but if it is our standard I'm ok with this.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,12 @@
 # CHANGELOG
 
 ## Not released
+
+- Client Side Widgets [#62](https://github.com/CartoDB/carto-react-lib/pull/62)
```
Comment (neokore) [inline CHANGELOG.md:5]: Why is this line added in this PR? I don't see anything related to CSW here.

File: .prettierrc (hunk 2 above)
Comment (AdriSolid) [reply .prettierrc:7]: I prefer to set `trailingCommas` to `none` as well (more like a personal preference)

File: package.json [hunk 5]
```diff
@@ -41,13 +42,29 @@
     "@turf/bbox-polygon": "^6.0.1",
     "@turf/boolean-contains": "^6.3.0",
     "@turf/boolean-intersects": "^6.0.2",
+    "@typescript-eslint/eslint-plugin": "^4.14.1",
+    "@typescript-eslint/parser": "^4.14.1",
```
Comment (neokore) [inline package.json:46]: I would like to remove this `eslint-config-react-app` dependencies, but @AdriSolid already tried and it fails in case of not adding them.
What about the plans of moving to typescript :stuck_out_tongue: ?

File: CHANGELOG.md (hunk 4 above)
Comment (AdriSolid) [reply CHANGELOG.md:5]: It was missed in last PR.

File: package.json [hunk 6]
```diff
@@ -58,5 +75,20 @@
...
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.+(js|jsx)": [
+      "yarn lint:fix",
+      "git add"
```
Comment (neokore) [inline package.json:87]: Just to confirm, we want this to be automatic, right? In other projects I usually get notified, but it doesn't fix and add changes automatically; but I understand that this "more comfortable" for devs.

File: package.json (hunk 5 above)
Comment (AdriSolid) [reply package.json:46]: The reason is mainly to have the ESLint config aligned with the template. So `eslint-config-react-app` needs TS packages..
I agree moving to TS! :laughing:

File: package.json (hunk 6 above)
Comment (AdriSolid) [reply package.json:87]: Indeed, we can open a discussion for that, but, it's using same config as template.

File: src/widgets/HistogramWidget.js [hunk 7]
```diff
@@ -8,21 +8,21 @@ import { getHistogram } from './models';
...
+ * @param  props
+ * @param  {string} props.id - ID for the widget instance.
+ * @param  {string} props.title - Title to show in the widget header.
+ * @param  {string} props.dataSource - ID of the data source to get the data from.
+ * @param  {string} props.column - Name of the data source's column to get the data from.
+ * @param  {string} props.operation - Operation to apply to the operationColumn. Must be one of those defined in `AggregationTypes` object.
+ * @param  {number[]} props.ticks - Array of thresholds for the X axis.
+ * @param  {formatterCallback} [props.xAxisformatter] - Function to format X axis values.
+ * @param  {formatterCallback} [props.formatter] - Function to format Y axis values.
+ * @param  {boolean} [props.viewportFilter=false] - Defines whether filter by the viewport or not.
```
Comment (neokore) [inline src/widgets/HistogramWidget.js:21]: I'm not sure this improves the definition for this parameter. I suggest to keep "globally" or explain a bit the other option.

Comment (AdriSolid) [reply src/widgets/HistogramWidget.js:21]: 'globally' LGTM

File: CHANGELOG.md (hunk 4 above)
Comment (neokore) [reply CHANGELOG.md:5]: I rather having a separate PR for this kind of small mistakes that can be merged quickly to keep everything clear and tidy, and avoid losing this item in changelog in case of this PR is not merged before releasing a new version. But if we are sure this PR is going to be merged soon before releasing, it can stay for this time.

Comment (AdriSolid) [reply CHANGELOG.md:5]: Not sure if it will be merged soon.. So let's remove it for the moment, and will create a small PR for that :+1:

File: package.json (hunk 5 above)
Comment (neokore) [reply package.json:46]: Ok, having this config synced with the one on the template is a priority.

File: package.json (hunk 6 above)
Comment (neokore) [reply package.json:87]: Agree, same as above.

**Review by neokore — APPROVED** [review]
LGTM, fix the yarn.lock conflict and we are ready to go
